src/main/java/com/conveyal/r5/streets/EdgeStore.java

```diff
@@ -728,6 +729,32 @@ public boolean canTurnFrom(StreetRouter.State s0, StreetRouter.State s1,
             return true;
         }
 
+        /**
+         * Cut the packed list of intermediate coordinates in two after specified segment
+         * @return List with first element int[] of coordinates before end of specified segment, and second element
+         * int[] of coordinates after end of specified segment.
+         */
+        public ArrayList<int[]> splitGeometryAfter(int segment) {
+            // Original packed coordinates of edge
+            int[] original = geometries.get(pairIndex);
+            ArrayList<int[]> geoms = new ArrayList<>();
+            geoms.add(Arrays.copyOfRange(original, 0, segment * 2)); // Multiply by 2 because packed array
+            geoms.add(Arrays.copyOfRange(original, segment * 2, original.length + 1));
+            return geoms;
+        }
+
+        /**
+         * Set intermediate coordinates directly with packed array
+         * @param coordinates Packed lists of lat, lon, lat, lon... as fixed-point integers
+         */
+        public void setGeometry (int[] coordinates) {
```

Comment:
There will be cases where one or both of the split geometry arrays is empty. We can save a bit of memory by detecting this and replacing it with the shared `EMPTY_INT_ARRAY` in this method. This optimization can also be mentioned in the Javadoc if added.